app/core/Tracker2/CookieConsentTracker.js

```diff
@@ -0,0 +1,74 @@
+import cookieconsent from 'cookieconsent'
+import 'cookieconsent/build/cookieconsent.min.css'
+
+import BaseTracker from './BaseTracker'
+
+/**
+ * Not a true tracker in that this tracker does not report data but instead
+ * prompts the user to consent to tracking.  It shares the same initialization
+ * and event emitter characteristics of a tracker.
+ */
+export default class CookieConsentTracker extends BaseTracker {
+  constructor (store) {
+    super()
+
+    this.store = store
+    this.loadStatus(undefined)
```

Comment:
Had to trace the code to see that this sets a default state of `false`. Very slick but wonder if the data model for status is a little complicated even with 3 values. How many states can it be in and what are the benefits of having this structure.

It looks like it can be either answered or not, and if it is answered it is in 3 states: "allow", "dismiss", or "deny". Is there a case against storing this string result? Thank you!

Reply:
This is a great call out that I unfortunately don't have an answer to as I was also confused by the states.  The goal with touching this part of the refactor was to extract the functionality for cookie consent into it's own tracker but I didn't focus on improving the logic here because I didn't want to introduce changes to cookie logic (which I'm assuming is critical for our GDPR compliance) without understanding all of the cases.  I wasn't able to understand it full and didn't want to risk attaching this type of logic change to a refactor PR.   We could add a comment here with a TODO to revisit and / or an Asana task?

Reply:
That makes total sense! I am up for leaving the logic as is with maybe just a todo comment.